### PR TITLE
feat(curriculum): restructured expense tracker python project

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52570

### CODE REVIEW

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65822308ded1ba4632f66aa3.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65822308ded1ba4632f66aa3.md:10]:
```suggestion
A dictionary is another built-in data type in Python. A dictionary is a collection of data in the form of *key*-*value* pairs. Dictionaries are defined with curly braces `{}` and they contain key-value pairs separated by commas. Each key is followed by a colon `:` and the corresponding value:
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/6582324b5d37fe4b622bc3c0.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/6582324b5d37fe4b622bc3c0.md:16]:
```suggestion
In the example above, `x` is a parameter to be used in the expression `expr`. Create a `test` variable and assign it a lambda function that takes an `x` parameter and returns `x * 2`.
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/6582352cbdbcde4d0c4f7b0b.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/6582352cbdbcde4d0c4f7b0b.md:10]:
```suggestion
To call a lambda function, you can use the usual function syntax with a pair of parentheses after the variable name.
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65823634a308cb4d6bcc5fa6.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65823634a308cb4d6bcc5fa6.md:10]:
```suggestion
Lambda functions can be effectively combined with the `map()` function, which executes a specified function for each element in a collection of objects, such as a list:
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65823634a308cb4d6bcc5fa6.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65823634a308cb4d6bcc5fa6.md:17]:
```suggestion
Modify your `print()` call to print the result of calling `map()` with `test` as the first argument and `[2, 3, 5, 8]` as the second argument.
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/658238f7604f154ea9a23e1e.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/658238f7604f154ea9a23e1e.md:12]:
```suggestion
For now, modify your current `print()` call by passing `sum(map(test, [2, 3, 5, 8]))` as the argument for testing.
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65823dde36392f505a39f7c7.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65823dde36392f505a39f7c7.md:10]:
```suggestion
Finally, pass your `map()` call to the `sum()` function to obtain the total expenses and return the result.
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65824111a09164518320088d.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/65824111a09164518320088d.md:16]:
```suggestion
`filter()` returns an iterator that includes the elements of `my_list` for which `my_function` returns `True`. An iterator is a special object that enables you to iterate over the elements of a collection, like a list.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Comment by ShaunSHamilton** [comment]
Please no one make any commits to this until I am done. It has to be restructured to fit with the `upcoming-python -> scientific-computing` move

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/6582507654b3ed5712341382.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/6582507654b3ed5712341382.md:12]:
```suggestion
Still in the `while` loop, under the `choice` variable, write an `if` statement to check if `choice` equals the string `1`. If it's true, this will be the starting point for adding a new expense.
```

File: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/658252f6b1526d57b103d48a.md
Comment (Sembauke) [inline curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-an-expense-tracker/658252f6b1526d57b103d48a.md:10]:
```suggestion
Inside the `if` statement, prompt the user to enter the amount for the expense and store it in a variable.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Review by Sembauke — APPROVED** [review]

**Comment by gikf** [comment]
Please remember about https://github.com/freeCodeCamp/freeCodeCamp/pull/52570#discussion_r1432280440 :heart:

**Comment by naomi-lgbt** [comment]
> Please remember about https://github.com/freeCodeCamp/freeCodeCamp/pull/52570#discussion_r1432280440 ❤️
Is that something we can address in a future PR? This is the last project I'm waiting on to unblock a few things :3

**Comment by ShaunSHamilton** [comment]
> > Please remember about [#52570 (comment)](https://github.com/freeCodeCamp/freeCodeCamp/pull/52570#discussion_r1432280440) ❤️
> 
> Is that something we can address in a future PR? This is the last project I'm waiting on to unblock a few things :3

Too late

File: curriculum/challenges/_meta/learn-lambda-functions-by-building-an-expense-tracker/meta.json
Comment (naomi-lgbt) [inline curriculum/challenges/_meta/learn-lambda-functions-by-building-an-expense-tracker/meta.json:7]:
```suggestion
  "order": 66,
```

Set this to something random for now - it'll prevent steps from conflicting in the tests.

I'll confirm the order in my CI PR after this lands. :3

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/_meta/learn-lambda-functions-by-building-an-expense-tracker/meta.json
Comment (ShaunSHamilton) [reply curriculum/challenges/_meta/learn-lambda-functions-by-building-an-expense-tracker/meta.json:7]: Ok.

**Comment by ShaunSHamilton** [comment]
That is me done reviewing this. It works locally.

**Review by ShaunSHamilton — APPROVED** [review]
Thanks for your work on this, Dario 🚀 

With another PR fixing the order and stuff, we can add an intro

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**
